h5py/_hl/group.py

```diff
@@ -134,10 +134,20 @@ def create_dataset(self, name, shape=None, dtype=None, data=None, **kwds):
             kwds['track_order'] = h5.get_config().track_order
 
         with phil:
-            dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)
+            group = self
+            if name:
+                if '/' in name:
+                    h5objects = [obj for obj in name.split('/') if len(obj)]
+                    name = h5objects[-1]
+                    h5objects = h5objects[:-1]
+
+                    for new_group in h5objects:
+                        group = self.get(new_group) or group.create_group(new_group)
+
+                name = str.encode(name)
```

Comment:
I think we always do encoding with `self.e_(name)`. It will do the same in this case, but the consistency is useful if we ever need to change how that works.